### PR TITLE
Add Kubernetes Dashboard

### PR DESCRIPTION
**Pull request by nebilbenmabrouk**
- Add a task in the workflow **Kops_Install_K8S_In_AWS** that deploys and displays an endpoint for Kubernetes Dashboard.

### CODE REVIEW

File: InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml [hunk 1]
```diff
@@ -759,10 +710,107 @@ variables.put("SIGNAL_ACTION","")
...
+
+    sleep 10
+    ((ATTEMPTS++))
+done
+
+CLUSTER_NAME="$variables_K8S_CLUSTER_NAME"
+
+echo "https://$LB_EXTERNAL_ADDRESS" > $DASHBOARD_DIR/$CLUSTER_NAME-dashboard-address
+
+TOKEN=$($K8S_TOOLS kubectl -n kubernetes-dashboard create token admin-user --duration=2h)
```
Comment (mboussaa) [inline InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml:777]: 2h is short and is it possible to have non expiring token ? 
how can I do after 2h + 1min ?

File: InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml [hunk 2]
```diff
@@ -759,10 +710,107 @@ variables.put("SIGNAL_ACTION","")
...
+        break
+    fi 
+
+    sleep 10
+    ((ATTEMPTS++))
+done
+
+CLUSTER_NAME="$variables_K8S_CLUSTER_NAME"
+
+echo "https://$LB_EXTERNAL_ADDRESS" > $DASHBOARD_DIR/$CLUSTER_NAME-dashboard-address
```
Comment (mboussaa) [inline InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml:775]: what is the certificate for http S ?

File: InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml [hunk 3]
```diff
@@ -759,10 +710,107 @@ variables.put("SIGNAL_ACTION","")
...
+	LB_EXTERNAL_ADDRESS=$($K8S_TOOLS kubectl get svc -n kubernetes-dashboard kubernetes-dashboard -o=jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+    echo "Dashboard service load balancer address: $LB_EXTERNAL_ADDRESS" 
+    
+    if [ ! -z "$LB_EXTERNAL_ADDRESS" ] && [ "$LB_EXTERNAL_ADDRESS" != "none" ] && [ "$LB_EXTERNAL_ADDRESS" != "pending" ]; then
+        break
+    fi 
+
+    sleep 10
+    ((ATTEMPTS++))
+done
```
Comment (mboussaa) [inline InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml:771]: how do you handle the error when the cluster cannot be created for some reasons ?

**Review by mbenguig — APPROVED** [review]

File: InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml (hunk 1 above)
Comment (nebilbenmabrouk) [reply InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml:777]: For security reasons it is not recommended to have a non-expring token.
The workflow will be mostly used for demos, that is why I set the token duration to 2h.
For long running clusters (more than 2 hours), we  can generate a new token via SSH or via any local kubectl client.

File: InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml (hunk 2 above)
Comment (nebilbenmabrouk) [reply InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml:775]: For now, the default certificate generated by the cert manager of Kubernetes. Investigating the SSL certificates is the next steps :)

**Comment by mboussaa** [comment]
do you provide a signal to terminate the cluster?

File: InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml (hunk 3 above)
Comment (nebilbenmabrouk) [reply InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml:771]: If this task fails, only the dashboard will be not deployed.
The admin can simply mark this task as finished and resume the workflow.
If the cluster deployment fails, we use the signal delete cluster to remove all the deployed resources.

File: InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml (hunk 2 above)
Comment (mboussaa) [reply InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml:775]: ok

File: InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml (hunk 3 above)
Comment (mboussaa) [reply InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml:771]: well I don't get it, I need to see it live to understand

**Comment by nebilbenmabrouk** [comment]
> @nebilbenmabrouk do you provide a signal to terminate the cluster? same question for update, like node-count

Yes I provide a signal to terminate the cluster. For now, updating the node count can be done the admin of the cluster, not via signals. The goal of the workflow is to show the deployment of K8S while reducing the resource consumption.

**Review by mboussaa — APPROVED** [review]

File: InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml (hunk 1 above)
Comment (nebilbenmabrouk) [reply InfrastructureAsCode/resources/catalog/Kops_Install_K8S_In_AWS.xml:777]: Yes, done.

**Pull request merged**